tests/tests/test_host_kdump.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/bash
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3, as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+if [ ! -v DEVICE_IP ]; then
+    DEVICE_IP="192.168.102.125"
+fi
+
+echo "Installing linux-crashdump and rebooting"
+ssh ubuntu@$DEVICE_IP sudo apt install linux-crashdump
+if [ $? != 0 ]; then
+    echo "Can't install linux-crashdump"
+    exit -1
+fi
+
+ssh ubuntu@$DEVICE_IP sudo systemctl reboot
+if [ $? != 0 ]; then
+    echo "Can't reboot"
+    exit -2
+fi
+
+sleep 30
+echo "Waiting for system to come back up"
+
+cnt=0
+until ssh ubuntu@$DEVICE_IP ls &> /dev/null; do sleep 1; cnt=$(expr $cnt + 1); if [ $cnt -gt 120 ]; then break; fi; done
+if [ $cnt -gt 60 ]; then
+    echo "Timed out waiting for $DEVICE_IP to come back up ($cnt)"
+    exit -3
+fi
+if [ $cnt == 0 ]; then
```

Comment:
Just curious why this is needed? Have you seen the system boot up quickly and that being indicative of a problem?

Reply:
The whole point of these tests is to ensure reboots, kexec, and crash (with kdump/kexec) work properly.   So, we want to make sure it actually rebooted/crashed, ....  If it doesn't actually reboot/crash/..., then its not really testing this properly.  So, its a failure if it doesn't actually reboot.

Reply:
Ah I see, so we're expecting the machine to still be in the process of rebooting after 15 seconds (or 10 seconds in other cases below). I imagine this generally works well but I'm thinking about some corner cases in the future involving servers with super fast boot disks or that use some sort of in-memory hibernation for fast rebooting.  

I wonder if testing this with commands that directly track and report reboot details may be more reliable. I Google'd around and [this thread](https://unix.stackexchange.com/questions/747014/command-to-check-if-the-machine-was-rebooted) mentions a few methods (`journalctl --list-boots` and `systemctl show --property=ActiveEnterTimestamp systemd-udevd.service`), although I don't know how they behave on crashes.